Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -433,8 +512,60 @@ public void OnExplosion(ExplosionEventArgs eventArgs)
                 _ => throw new ArgumentOutOfRangeException()
             };
 
-            ChangeDamage(GetDamageType("Piercing"), damage, false);
-            ChangeDamage(GetDamageType("Heat"), damage, false);
+            foreach (var typeID in ExplosionDamageTypeIDs)
+            {
+                ChangeDamage(_prototypeManager.Index<DamageTypePrototype>(typeID), damage, false);
+            }
+        }
+
+        // TODO QUESTION I created this function, and the one below it, to covert between Dictionary<IPrototype,int> and
```

Comment:
Does a function like this already exist somewhere? Should this function be here?

Reply:
if it stays this generalized it should go in PrototypeManager instead, otherwise just use it wherever the one place is

Reply:
Well if sending stuff over the network continues to need strings instead of prototypes, I think it will stay usefull? So ill move it to prototype manager